.github/workflows/documentation-tests.yaml

```
@@ -84,62 +77,40 @@ jobs:
-            RESPONSE_CODE=$(git ls-remote --heads $DOCS_REPO $TEMP_BRANCH | wc -l)
-
-            if [[ ${RESPONSE_CODE} == "0" ]]
-            then
-              echo "PR source branch '$TEMP_BRANCH' branch DOES NOT exist in $DOCS_REPO"
-
-              # Attempt to use PR's base branch
-              TEMP_BRANCH=${{ github.base_ref }}
-              RESPONSE_CODE=$(git ls-remote --heads $DOCS_REPO $TEMP_BRANCH | wc -l)
-
-              if [[ ${RESPONSE_CODE} == "0" ]]
-              then
-                echo "PR base branch '$TEMP_BRANCH' branch DOES NOT exist in $DOCS_REPO"
-
-                # Attempt to use version's branch, which will most likely be the base anyways.
-                # This is just in case the PR was to an intermediate branch
-                TEMP_BRANCH=master
-                RESPONSE_CODE=$(git ls-remote --heads $DOCS_REPO $TEMP_BRANCH | wc -l)
-
-                if [[ ${RESPONSE_CODE} == "0" ]]
-                then
-                  # There are no more fallbacks, so we need to fail here
-                  echo "Version branch '$TEMP_BRANCH' branch DOES NOT exist in $DOCS_REPO"
-                  exit 1
-                fi
-              fi
-            fi
-
-            echo "ACTION_BRANCH_NAME=$TEMP_BRANCH" >> $GITHUB_ENV
-            echo "PR event: using deduced branch '$TEMP_BRANCH' for docs repository"
-          fi
-
-      - name: Download FastDDS documentation repo
+      - name: Fetch Fast DDS CI dependencies
+        uses: eProsima/eProsima-CI/multiplatform/vcs_import@v0
+        with:
+          vcs_repos_file: ${{ github.workspace }}/src/fastrtps/.github/workflows/config/ci.repos
+          destination_workspace: src
+          skip_existing: 'true'
+
+      - name: Get Fast DDS Python branch
+        id: get_fastdds_python_branch
+        uses: eProsima/eProsima-CI/ubuntu/get_related_branch_from_repo@feature/get_related_branch_from_repo
+        with:
+          remote_repository: eProsima/Fast-DDS-python
+          fallback_branch: ${{ inputs.fastdds_python_branch || 'main' }}
+
+      - name: Download Fast DDS Python repo
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          repository: eProsima/Fast-DDS-python
+          path: src/fastdds-python
+          ref: ${{ steps.deduce_branch.outputs.deduced_branch }}
+
+      - name: Get Fast DDS Docs branch
+        id: get_fastdds_docs_branch
+        uses: eProsima/eProsima-CI/ubuntu/get_related_branch_from_repo@feature/get_related_branch_from_repo
+        with:
+          remote_repository: eProsima/Fast-DDS-docs
+          fallback_branch: ${{ inputs.fastdds_docs_branch || 'master' }}
+
+      - name: Download Fast DDS documentation repo
         uses: eProsima/eProsima-CI/external/checkout@v0
         with:
           repository: eProsima/Fast-DDS-docs
           path: src/fastdds-docs
-          ref: ${{ env.ACTION_BRANCH_NAME }}
+          ref: ${{ steps.deduce_branch.outputs.deduced_branch }}
```

As before, I think this would need to use the step ID

```suggestion
          ref: ${{ steps.get_fastdds_docs_branch.outputs.deduced_branch }}
```

---

I agree, because it correctly identify the branch to use in the step "Get Fast DDS Docs branch" but it doesn't seem to use it in the following step "Download Fast DDS documentation repo"